worker/include/RTC/RtpPacket.hpp

```diff
@@ -131,8 +131,89 @@ namespace RTC
 			);
 			// clang-format on
 		}
+		class SharedPtr
```

Comment:
Why is there a custom shared pointer implementation instead of using `std::shared_pointer`?

Reply:
The standard one will call `delete` and de-allocate memory. This implementation will call destructor only and return memory to the pool to be consumed by the next packet. All packets are of the same size, so there is no need to re-allocate them all the time.

Reply:
@nazar-pc you can pass custom deleter into shared_ptr constructor

Reply:
`
std::shared_ptr<int> pIntPtr(new int(10), 
    [](int *pi) { delete pi; })
    `

Reply:
Correct,

In `RtpPacket::Clone()`:
```c++
     auto* packet = RtpPacketPool.Allocate();
     new (packet) RtpPacket(header, headerExtension, payload, payloadLength, payloadPadding, len);

     std::shared_ptr<RtpPacket> shared(packet, /*Deleter*/ [](RtpPacket* packet) {
         // Call destructor manually since memory was pre-allocated upfront.
	 packet->~RtpPacket();
	 // Return packet into object pool for future reuse of memory allocation.
	 RtpPacketPool.Return(packet);
     });
```

Reply:
@nazar-pc, I will create a PR for this one.

Reply:
There are several places where pools are used, I think we need to apply this patter in all cases.